arch/riscv/Kconfig

```diff
@@ -37,6 +37,12 @@ config INCLUDE_RESET_VECTOR
 	  Include the reset vector stub, which initializes the stack and
 	  prepares for running C code.
 
+config RISCV_GLOBAL_POINTER
```

Comment:
Is there a reason to make this a Kconfig vs just always supporting it?

Reply:
Yes,
Using RISC-V global pointer is optional and currently there no RISC-V vendors SoC which are using it.
To use global pointer feature the "__global_pointer$" variable should be defined in linker script.

With this solution global pointer usage is disabled by default and it is backward compatible with all other RISC-V SoC vendors.

Reply:
However, if we add "__global_pointer$" variable to the generic RISC-V linker script, it will be supported by all RISC-V vendors automatically and in this case we can make it always supporting.
But, I still preferer to have it user configurable because usage of RISC-V global pointer is optional.

Reply:
Would be good to pull this commit into its own PR to allow review by RISC-V collaborators.